From 44483ea0b5aa78fb9d449966b158c1ed4c3a9b18 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Thu, 14 Oct 2021 17:37:22 +0800
Subject: [PATCH] sched: Introcude config option SCHED_OPTIMIZE_LOAD_TRACKING
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 44483ea0b5aa78fb9d449966b158c1ed4c3a9b18


hulk inclusion
category: feature
bugzilla: 175551 https://gitee.com/openeuler/kernel/issues/I4DVI3

--------------------------------

Introduce kernel config SCHED_OPTIMIZE_LOAD_TRACKING to prepare
for kernel scheduler loadtracking optimize.

Signed-off-by: Yu Jiahua <yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 init/Kconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index b095baa2b83c..66d7f2708b95 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2077,6 +2077,14 @@ endmenu		# General setup
 
 source "arch/Kconfig"
 
+config SCHED_OPTIMIZE_LOAD_TRACKING
+	bool "Optimize scheduler load tracking"
+	default n
+	help
+	  Optimize scheduler load tracking, when load balance is not important
+	  in system, we close some load tracking in tick and enqueue or dequeue
+	  task, in this way, we can save some unnecessary cpu overhead.
+
 config RT_MUTEXES
 	bool
 
-- 
2.26.2

